### PR TITLE
enable optimizer

### PR DESCRIPTION
**Pull request by daviszhen**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

The frontend will call the optimizer to optimize the select statement.

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/frontend/mysql_cmd_executor.go [hunk 1]
```diff
@@ -1551,9 +1551,23 @@ func (cwft *TxnComputationWrapper) GetAffectedRows() uint64 {
 
 func (cwft *TxnComputationWrapper) Compile(u interface{}, fill func(interface{}, *batch.Batch) error) (interface{}, error) {
 	var err error
-	cwft.plan, err = plan2.BuildPlan(cwft.ses.GetTxnCompilerContext(), cwft.stmt)
-	if err != nil {
-		return nil, err
+	switch stmt := cwft.stmt.(type) {
```
Comment (nnsgmsone) [inline pkg/frontend/mysql_cmd_executor.go:1554]: ParenSelect Optimiser also required

**Review by nnsgmsone — COMMENTED** [review]
explain also requires  call  optimiser.

File: pkg/frontend/mysql_cmd_executor.go (hunk 1 above)
Comment (daviszhen) [reply pkg/frontend/mysql_cmd_executor.go:1554]: I will add it.

Comment (nnsgmsone) [reply pkg/frontend/mysql_cmd_executor.go:1554]: Delete, Update, Insert also required to call optimizer

Comment (daviszhen) [reply pkg/frontend/mysql_cmd_executor.go]: I have added it.

**Review by nnsgmsone — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Pull request merged**